src/test/java/teammates/test/cases/webapi/GetStudentProfilePictureActionTest.java

```diff
@@ -29,57 +26,195 @@ protected String getRequestMethod() {
 
-        String[] submissionParams = new String[] {};
+        GetStudentProfilePictureAction action = getAction();
+        ImageResult imageResult = getImageResult(action);
 
-        GetStudentProfilePictureAction action = getAction(submissionParams);
-        assertThrows(NullHttpParameterException.class, () -> action.execute());
-    }
+        assertEquals(HttpStatus.SC_OK, imageResult.getStatusCode());
 
-    /**
-     * Tests the branch of the Action handling a request from the Student
-     * directly, where the parameters are simply the blobKey of the picture
-     * itself.
-     */
-    private void testActionWithBlobKey() {
-        testActionWithBlobKeySuccess();
-    }
-
-    private void testActionWithBlobKeySuccess() {
-        ______TS("Typical case: using blobkey");
-        loginAsStudent(account.googleId);
+        ______TS("Success case: student passes in incomplete params but still gets his own image");
 
         String[] submissionParams = new String[] {
-                Const.ParamsNames.BLOB_KEY, studentProfileAttributes.pictureKey,
+                Const.ParamsNames.COURSE_ID, student1InCourse1.getCourse(),
+        };
+
+        action = getAction(submissionParams);
+        imageResult = getImageResult(action);
+
+        assertEquals(HttpStatus.SC_OK, imageResult.getStatusCode());
+
+        submissionParams = new String[] {
+                Const.ParamsNames.STUDENT_EMAIL, student1InCourse1.getEmail(),
+        };
+
+        action = getAction(submissionParams);
+        imageResult = getImageResult(action);
+
+        assertEquals(HttpStatus.SC_OK, imageResult.getStatusCode());
+
+        ______TS("Success case: student gets his teammate's image");
+        StudentAttributes student2InCourse1 = typicalBundle.students.get("student2InCourse1");
+        gaeSimulation.logoutUser();
+        loginAsStudent(student2InCourse1.googleId);
+
+        submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, student1InCourse1.getCourse(),
+                Const.ParamsNames.STUDENT_EMAIL, student1InCourse1.getEmail(),
+        };
+
+        action = getAction(submissionParams);
+        imageResult = getImageResult(action);
+
+        assertEquals(HttpStatus.SC_OK, imageResult.getStatusCode());
+
+        ______TS("Success case: instructor with privilege views image of his student");
+        gaeSimulation.logoutUser();
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        loginAsInstructor(instructor1OfCourse1.googleId);
+
+        submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, student1InCourse1.getCourse(),
+                Const.ParamsNames.STUDENT_EMAIL, student1InCourse1.getEmail(),
+        };
+
+        action = getAction(submissionParams);
+        imageResult = getImageResult(action);
+
+        assertEquals(HttpStatus.SC_OK, imageResult.getStatusCode());
```

Comment:
Is there a way to verify the image is correct as well?

Reply:
Yup done!